sections/errorhandling/shuttingtheprocess.md

```diff
@@ -22,12 +24,56 @@ function errorHandler() {
       .then(saveInOpsQueueIfCritical)
       .then(determineIfOperationalError);
   }
-
+<br/>
   this.isTrustedError = function (error) {
     return error.isOperational;
   }
 }
-```
+</pre>
+</details>
+
+<details>
+<summary><strong>Typescript</strong></summary>
+<pre lang="typescript">
+// Assuming developers mark known operational errors with error.isOperational=true, read best practice #3
+process.on('uncaughtException', (error: Error) => {
+  errorManagement.handler.handleError(error);
+  if(!errorManagement.handler.isTrustedError(error))
+    process.exit(1)
+});
+<br/>
+// centralized error object that derives from Node’s Error
+export class AppError extends Error {
+  public readonly isOperational: boolean;
+<br/>
+  constructor(description: string, isOperational: boolean) {
+    super(description);
+    Object.setPrototypeOf(this, new.target.prototype); // restore prototype chain
+    this.isOperational = isOperational;
+    Error.captureStackTrace(this);
+  }
+}
+<br/>
+// centralized error handler encapsulates error-handling related logic
+class ErrorHandler {
+  public async handleError(err: Error): Promise<void> {
+    await logger.logError(err);
+    await sendMailToAdminIfCritical();
+    await saveInOpsQueueIfCritical();
+    await determineIfOperationalError();
+  };
+<br/>
+  public isTrustedError(error: Error) {
+    if (error instanceof AppError) {

```

Comment:
Given that we cannot have `isOperational` on `Error` type, I added this check.

Reply:
That's definitely the right path here 😄